backends/ebpf/ebpfType.h

```diff
@@ -39,6 +39,7 @@ class EBPFType : public EBPFObject {
     virtual void declareArray(CodeBuilder * /*builder*/, cstring /*id*/, unsigned /*size*/) {
         BUG("%1%: unsupported array", type);
     }
+    virtual bool is_array() { return (false); }
```

Comment:
Nit: Shouldn't you be able to tell by the type of the value? Cast to a scalar value, then check the width.

Reply:
Fixed